core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt

```diff
@@ -1,68 +1,93 @@
 package com.amplitude.core.utilities
 
+import com.amplitude.common.Logger
 import com.amplitude.id.utilities.KeyValueStore
 import com.amplitude.id.utilities.createDirectory
+import kotlinx.coroutines.runBlocking
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
 import java.io.BufferedReader
 import java.io.File
+import java.io.FileNotFoundException
 import java.io.FileOutputStream
+import java.io.IOException
 import java.util.Collections
+import java.util.Random
 import java.util.concurrent.ConcurrentHashMap
 
 class EventsFileManager(
     private val directory: File,
     private val storageKey: String,
-    private val kvs: KeyValueStore
+    private val kvs: KeyValueStore,
+    private val logger: Logger,
 ) {
-    init {
-        createDirectory(directory)
-    }
-
     private val fileIndexKey = "amplitude.events.file.index.$storageKey"
+    private val storageVersionKey = "amplitude.events.file.version.$storageKey"
+    val filePathSet: MutableSet<String> = Collections.newSetFromMap(ConcurrentHashMap<String, Boolean>())
+    val curFile: MutableMap<String, File> = ConcurrentHashMap<String, File>()
+    private val diagnostics = Diagnostics()
```

Comment:
Diagnostics feels a little generic. Does it make sense for diagnostics to be scoped to the SDK instead of scoped to the EventsFileManager? I'd imagine we can collect diagnostics from other places too and not just within EventsFileManager. 

If you don't want scope creep and want to just focus on malformed events and just collect that here.

Reply:
I am still debating about this and would like to gather the thoughts. Put here for the first iteration and reuse the mutex from the EventsFileManager, though I am considering if I should move this out to make it available to all.

Reply:
Migrated this on the amplitude instance level so it could possibly cover more if needed.